api/metric/common.go

```diff
@@ -65,6 +65,10 @@ func (bm *baseMetric) base() *baseMetric {
 	return bm
 }
 
+func (bm *baseMetric) DefinitionID() core.EventID {
```

Comment:
should this be named `EventID()`?

Reply:
core.EventID here is used as Metric definition ID. It is really not an event. However, it is confusing for sure. 
@jmacd what is your opinion?

Reply:
Suggestion: Godoc documentation would be ++

Reply:
Is it possible to rename the struct value to `definitionID core.EventID`?